Stratis.Bitcoin/Features/Wallet/HdOperations.cs

```diff
@@ -0,0 +1,211 @@
+        /// <returns>
+        /// An HD public key derived from an extended public key.
+        /// </returns>
+        public static PubKey GeneratePublicKey(string accountExtPubKey, int index, bool isChange)
+        {
+            Guard.NotEmpty(accountExtPubKey, nameof(accountExtPubKey));
+
+            int change = isChange ? 1 : 0;
+            KeyPath keyPath = new KeyPath($"{change}/{index}");
+            ExtPubKey extPubKey = ExtPubKey.Parse(accountExtPubKey).Derive(keyPath);
+            return extPubKey.PubKey;
+        }
+
+        /// <summary>
+        /// Gets the extended private key for an account.
+        /// </summary>
+        /// <param name="privateKey">The private key from which to generate the extended private key.</param>
+        /// <param name="chainCode">The chain code used in creating the extended private key.</param>
+        /// <param name="hdPath">The HD path of the account for which to get the extended private key.</param>
+        /// <param name="network">The network for which to generate this extended private key.</param>
+        /// <returns></returns>
+        public static ISecret GetExtendedPrivateKey(Key privateKey, byte[] chainCode, string hdPath, Network network)
+        {
+            Guard.NotNull(privateKey, nameof(privateKey));
+            Guard.NotNull(chainCode, nameof(chainCode));
+            Guard.NotEmpty(hdPath, nameof(hdPath));
+            Guard.NotNull(network, nameof(network));
+
+            // Get the extended key.
+            ExtKey seedExtKey = new ExtKey(privateKey, chainCode);
+            ExtKey addressExtKey = seedExtKey.Derive(new KeyPath(hdPath));
+            BitcoinExtKey addressPrivateKey = addressExtKey.GetWif(network);
+            return addressPrivateKey;
+        }
+
+        /// <summary>
+        /// Gets the extended public key for an account.
+        /// </summary>
+        /// <param name="privateKey">The private key from which to generate the extended public key.</param>
+        /// <param name="chainCode">The chain code used in creating the extended public key.</param>
+        /// <param name="coinType">Type of the coin of the account for which to generate an extended public key.</param>
+        /// <param name="accountIndex">Index of the account for which to generate an extended public key.</param>
+        /// <returns>The extended public key for an account, used to derive child keys.</returns>
+        public static ExtPubKey GetExtendedPublicKey(Key privateKey, byte[] chainCode, int coinType, int accountIndex)
+        {
+            Guard.NotNull(privateKey, nameof(privateKey));
+            Guard.NotNull(chainCode, nameof(chainCode));
+
+            var accountHdPath = GetAccountHdPath(coinType, accountIndex);
+            return GetExtendedPublicKey(privateKey, chainCode, accountHdPath);
+        }
+
+        /// <summary>
+        /// Gets the extended public key for an account.
```

Comment:
yeah but it's the hd path of an account.
Maybe Gets the extended public key for an account corresponding to an HD path.